### PR TITLE
Add time and period to update function

### PR DESCRIPTION
**Pull request by bmagyar**
Following up https://github.com/ros-controls/ros2_control/pull/520

### CODE REVIEW

File: diff_drive_controller/src/diff_drive_controller.cpp [hunk 1]
```diff
@@ -145,7 +146,7 @@ controller_interface::return_type DiffDriveController::update()
     return controller_interface::return_type::OK;
   }
 
-  const auto current_time = node_->get_clock()->now();
+  const auto current_time = time;
```
Comment (destogl) [inline diff_drive_controller/src/diff_drive_controller.cpp:149]: In line 159 on e should use `period`

**Review by destogl — APPROVED** [review]
Looks good for the first port. Do not forget open a follow-up issue where the paramters should be used by the controllers.

File: diff_drive_controller/src/diff_drive_controller.cpp (hunk 1 above)
Comment (bmagyar) [reply diff_drive_controller/src/diff_drive_controller.cpp:149]: that's not reeeeally `dt`, I'll rename the variable

File: diff_drive_controller/src/diff_drive_controller.cpp [hunk 2]
```diff
@@ -132,7 +132,8 @@ InterfaceConfiguration DiffDriveController::state_interface_configuration() cons
   return {interface_configuration_type::INDIVIDUAL, conf_names};
 }
 
-controller_interface::return_type DiffDriveController::update()
+controller_interface::return_type DiffDriveController::update(
+  const rclcpp::Time & time, const rclcpp::Duration & period)
```
Comment (v-lopez) [inline diff_drive_controller/src/diff_drive_controller.cpp:136]: Should we remove the `period` variable name so it doesn't complain about unused variables?
```suggestion
  const rclcpp::Time & time, const rclcpp::Duration & /*period*/)
```

File: forward_command_controller/src/forward_command_controller.cpp [hunk 3]
```diff
@@ -153,7 +153,8 @@ CallbackReturn ForwardCommandController::on_deactivate(
   return CallbackReturn::SUCCESS;
 }
 
-controller_interface::return_type ForwardCommandController::update()
+controller_interface::return_type ForwardCommandController::update(
+  const rclcpp::Time & time, const rclcpp::Duration & period)
```
Comment (v-lopez) [inline forward_command_controller/src/forward_command_controller.cpp:157]:
```suggestion
  const rclcpp::Time & time, const rclcpp::Duration & /*period*/)
```

File: gripper_controllers/include/gripper_controllers/gripper_action_controller_impl.hpp [hunk 4]
```diff
@@ -61,7 +61,8 @@ CallbackReturn GripperActionController<HardwareInterface>::on_init()
 }
 
 template <const char * HardwareInterface>
-controller_interface::return_type GripperActionController<HardwareInterface>::update()
+controller_interface::return_type GripperActionController<HardwareInterface>::update(
+  const rclcpp::Time & time, const rclcpp::Duration & period)
```
Comment (v-lopez) [inline gripper_controllers/include/gripper_controllers/gripper_action_controller_impl.hpp:65]:
```suggestion
  const rclcpp::Time & time, const rclcpp::Duration & /*period*/)
```

File: joint_state_broadcaster/src/joint_state_broadcaster.cpp [hunk 5]
```diff
@@ -229,7 +229,8 @@ double get_value(
   }
 }
 
-controller_interface::return_type JointStateBroadcaster::update()
+controller_interface::return_type JointStateBroadcaster::update(
+  const rclcpp::Time & time, const rclcpp::Duration & period)
```
Comment (v-lopez) [inline joint_state_broadcaster/src/joint_state_broadcaster.cpp:233]:
```suggestion
  const rclcpp::Time & time, const rclcpp::Duration & /*period*/)
```

File: joint_trajectory_controller/src/joint_trajectory_controller.cpp [hunk 6]
```diff
@@ -105,7 +105,8 @@ JointTrajectoryController::state_interface_configuration() const
   return conf;
 }
 
-controller_interface::return_type JointTrajectoryController::update()
+controller_interface::return_type JointTrajectoryController::update(
+  const rclcpp::Time & time, const rclcpp::Duration & period)
```
Comment (v-lopez) [inline joint_trajectory_controller/src/joint_trajectory_controller.cpp:109]:
```suggestion
  const rclcpp::Time & time, const rclcpp::Duration & /*period*/)
```

**Pull request merged**